mantidproject/mantid
# Embed instrument viewer in refl gui

**Description of work.**

This PR embeds the instrument viewer display in the preview tab on the ISIS Reflectometry GUI.
- An `InstViewModel` has been created to handle creating the instrument actor for the loaded workspace, and the surface to plot
- The `PreviewPresenter` has handles both the synchronous case - where the workspace already exists in the ADS and can be plotted immediately - and the async case where it needs to be loaded via an algorithm.
- `InstrumentActor` has been modified to take a message handler to handle pop-up messages rather than using QMessageBox itself. This is to avoid pop-up dialogs appearing when we are using `InstrumentActor` in unit tests.
- A change to `GLDisplay` was required to ensure the instrument view plot is blank when first created, rather than just rendering what was on screen previously

**To test:**

- Open the ISIS Reflectometry interface and set instrument to INTER
- Go to the Preview tab. The instrument viewer plot should be blank
- Enter a run number for an unloaded workspace e.g. `13460` and press enter or click Load
- The workspace should be displayed in the instrument plot
- Load a different workspace into mantid separately and give it a different name e.g. INTER45455
- Enter this name in the Preview tab and click load. This should use the workspace from the ADS rather than re-loading it. It should be displayed in the instrument view plot.

- Also test the basic functionality of instrument viewer to check that nothing is broken

Fixes #32490 





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
